pygam/pygam.py

```diff
@@ -87,22 +87,74 @@
              'diffs': Diffs,
              'accuracy': Accuracy,
              'coef': Coef
-            }
+             }
 
 PENALTIES = {'auto': 'auto',
              'derivative': derivative,
              'l2': l2,
              'none': none,
-            }
+             }
 
 CONSTRAINTS = {'convex': convex,
                'concave': concave,
                'monotonic_inc': monotonic_inc,
                'monotonic_dec': monotonic_dec,
                'circular': circular,
                'none': none
-              }
-
+               }
+
+
+def _parallel_inc(gam, X, y, weights):
+    n = len(y)
+    R = []
+    f = []
+    r = 0.
+    vars_ = defaultdict(list)
+    pool = Pool()
+    block_results = []
+    for mask in gam._block_masks(n):
+        Xk = gam._modelmat(X[mask])
+        block_results.append(pool.apply_async(process_one,
+                                              args=(gam, Xk, y[mask], weights[mask],)))
+    pool.close()
+    pool.join
+
+    res = [x.get() for x in block_results if x is not None]
+    res = list(filter(None.__ne__, res))
```

Comment:
this line is causing failure in python 2.x because `None` doesn't have the `__ne__` method before python 3.
why do we need this line anyways? 
is it possible x is not None, but still x.get() returns None?

either way, perhaps we should just do another list comprehension?

```python
results = [x.get() for x in block_results if x is not None]
results = [result for result in results if result is not None]
```

Reply:
@maorn

Reply:
these are general lines i use in all parallel methods for cases where you returned None in the function, in our case we always return a value.
so the second line can be removed